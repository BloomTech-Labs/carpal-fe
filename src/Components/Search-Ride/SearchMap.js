import React, { useState, useEffect } from "react";
import ReactMapGL, { Marker } from "react-map-gl";

import Logo from "./../../img/Logo.png";

import "./SearchRide.scss";
import "mapbox-gl/dist/mapbox-gl.css";
import { config } from "dotenv";
config();

const mapboxAPI = process.env.REACT_APP_MAPBOX_TOKEN;

function SearchMap() {
    const [viewport, setViewport] = useState({
        latitude: 0,
        longitude: 0,
        zoom: 15,
        width: "100%",
        height: "100vh",
        position: "center"
    });
    //State for keeping track of the Markers long/lat
    const [marker, setMarker] = useState({
        latitude: 0,
        longitude: 0,
    });
    useEffect(() => {
        // without this test will fail from not being able to run this function
            navigator.geolocation.getCurrentPosition(getUserLocation);
    
    }, []);
    console.log('Corsd')
    const getUserLocation = (position) => {
        var crd = position.coords;
       
        setViewport({
            ...viewport,
            latitude: crd.latitude,
            longitude: crd.longitude
        });
        //When we Get users location we set the marker to the users current location
        setMarker({
            latitude: crd.latitude,
            longitude: crd.longitude
        });
    };

    const handleDragEnd = (event) => {
        setViewport({
            ...viewport,
            longitude: event.lngLat[0],
            latitude: event.lngLat[1]
        });
        //Update Users location on dragend
        setMarker({
            longitude: event.lngLat[0],
            latitude: event.lngLat[1]
        });
    };
    return (
        <div role="map-wrapper" className="map">
            <ReactMapGL
                {...viewport}
                mapboxApiAccessToken={mapboxAPI}
                onViewportChange={(viewport) => {
                    setViewport(viewport);
                }}
            >
                <Marker
                    latitude={marker.latitude}
                    longitude={marker.longitude}
                    offsetLeft={-10}
                    offsetTop={-10}
                    draggable={true}
                    onDragEnd={handleDragEnd}
                >
                    <img
                        src={`${Logo}`}
                        alt="marker"
                        style={{
                            width: "20px",
                            height: "20px",
                            borderRadius: "50%"
                        }}
                    />
                </Marker>
            </ReactMapGL>
        </div>
    );
}


export default SearchMap;
