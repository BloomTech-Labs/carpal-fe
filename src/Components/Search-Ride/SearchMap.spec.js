import SearchMap from "./SearchMap";
import * as rtl from "@testing-library/react";
import React from "react";
import ReactDOM from "react-dom";
import "@testing-library/jest-dom/extend-expect";


afterEach(rtl.cleanup);

const mockGeolocation = {
    getCurrentPosition: jest.fn().mockImplementationOnce((vals) => vals)
};

//assign geolocation the mockGeolocation obj
navigator.geolocation = mockGeolocation;

describe("SearchMap", () => {
    test("Renders without crashing", () => {
        const div = document.createElement("div");

        ReactDOM.render(<SearchMap />, div);
    });

    test("Matches snapshot", () => {
        const wrapper = rtl.render(<SearchMap />);

        expect(wrapper).toMatchSnapshot();
    });
});