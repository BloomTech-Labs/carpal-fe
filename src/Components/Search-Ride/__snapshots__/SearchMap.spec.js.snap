// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchMap Matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="map"
        role="map-wrapper"
      >
        <div
          style="position: relative; width: 0px; height: 0px; cursor: grab; outline: none;"
          tabindex="0"
        >
          <div
            style="position: relative; width: 100%; height: 100%;"
          >
            <div
              class=""
              style="position: absolute; width: 100%; height: 100%; overflow: hidden; visibility: inherit;"
            />
            <div
              style="overflow: visible; height: 0px; width: 0px;"
            />
            <div
              class="resize-triggers"
            >
              <div
                class="expand-trigger"
              >
                <div
                  style="width: 1px; height: 1px;"
                />
              </div>
              <div
                class="contract-trigger"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="map"
      role="map-wrapper"
    >
      <div
        style="position: relative; width: 0px; height: 0px; cursor: grab; outline: none;"
        tabindex="0"
      >
        <div
          style="position: relative; width: 100%; height: 100%;"
        >
          <div
            class=""
            style="position: absolute; width: 100%; height: 100%; overflow: hidden; visibility: inherit;"
          />
          <div
            style="overflow: visible; height: 0px; width: 0px;"
          />
          <div
            class="resize-triggers"
          >
            <div
              class="expand-trigger"
            >
              <div
                style="width: 1px; height: 1px;"
              />
            </div>
            <div
              class="contract-trigger"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
